kmir/src/kmir/kdist/mir-semantics/kmir.k

```diff
@@ -14,167 +15,717 @@ module KMIR-SYNTAX
+  rule #makeAggregate(aggregateKindCoroutine(_CoroutineDef, _Generics, _Movability),                                 Args ) => #makeStruct( 0, Args )
+  rule #makeAggregate(aggregateKindRawPtr(_Ty, _Mutability),                                                         Args ) => #buildPtr( Args )
+
+  // Intrinsic
+
+  syntax KItem ::= #eval(NonDivergingIntrinsic)
+
+  rule <k> #operands(ARGS) ~> #eval(nonDivergingIntrinsicAssume(_))             => #assume( ARGS ) ... </k>
+  rule <k> #operands(ARGS) ~> #eval(nonDivergingIntrinsicCopyNonOverlapping(_)) => #copy( ARGS )   ... </k>
+
+  // Statement/Terminator Evaluation
+
+  syntax KItem ::= #exec(StatementKind)
+                 | #exec(TerminatorKind)
+
+  rule <k> basicBlock( statement(STMT, _SPAN) STMTS, TERM )     => #exec( STMT ) ~> basicBlock( STMTS, TERM ) ... </k>
+  rule <k> basicBlock( .Statements, terminator( TERM, _SPAN ) ) => #evalArgs( TERM ) ~> #exec( TERM )         ... </k>
+
+  // Statements
+
+  syntax KItem ::= #assign(Place)
+
+  rule <k> #exec(statementKindAssign(PLACE, RVAL))                           => #evalArgs(RVAL) ~> #eval(RVAL) ~> #assign(PLACE) ... </k>
+  rule <k> #value(VAL) ~> #assign(PLACE)                                     => #setPlace(PLACE, VAL)                            ... </k>
+  rule <k> #exec(statementKindIntrinsic(INT))                                => #evalArgs(INT) ~> #eval(INT)                     ... </k>
+  rule <k> #exec(statementKindSetDiscriminant(PLACE, variantIdx(IDX)))       => #setDisc(PLACE, IDX)                             ... </k>
+  rule <k> #exec(deinit(PLACE))                                              => #deinitPlace(PLACE)                              ... </k>
+  rule <k> #exec(statementKindFakeRead(_FakeReadCause, _Place))              => .K                                               ... </k>
+  rule <k> #exec(statementKindStorageLive(_Local))                           => .K                                               ... </k>
+  rule <k> #exec(statementKindStorageDead(_Local))                           => .K                                               ... </k>
+  rule <k> #exec(statementKindRetag(_RetagKind, _Place))                     => .K                                               ... </k>
+  rule <k> #exec(statementKindPlaceMention(_Place))                          => .K                                               ... </k>
+  rule <k> #exec(statementKindAscribeUserType(_Place, _UserType, _Variance)) => .K                                               ... </k>
+  rule <k> #exec(statementKindCoverage(_Coverage))                           => .K                                               ... </k>
+  rule <k> #exec(statementKindConstEvalCounter)                              => .K                                               ... </k>
+  rule <k> #exec(statementKindNop)                                           => .K                                               ... </k>
```

Comment:
Could you add comments next to the exec rules for which of these statements are actually NoOps (so the .K should remain) or TODOs?

Reply:
Yes, I will add a comment about that.